### ISSUE TITLE
fix: reject invalid rule option values at startup instead of silently falling back

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#203]
## Overview

When a rule option receives an invalid value (e.g. `"style": "hoge"` for `consistent-code-fence`), gomarklint silently falls back to a default and continues running. The user gets no feedback that their config is wrong.

## Problem

Rules that accept a `style` option (`consistent-code-fence`, `consistent-emphasis-style`, `consistent-list-marker`) validate the value inside their linter helper and fall back to `"consistent"` when the value is unrecognised:

```go
switch style {
case "consistent", "backtick", "tilde":
    return style
default:
    return "consistent" // silently ignores invalid input
}
```

A user who mistyped `"style": "backticks"` will never know — the linter just runs with `"consistent"` instead.

## Proposed Fix

Validate rule option values at startup and exit with a clear error message before any linting runs. For example:

```
gomarklint: invalid value "hoge" for consistent-code-fence.style (valid values: consistent, backtick, tilde)
```

## Scope

Rules affected:
- `consistent-code-fence` (style: `consistent` | `backtick` | `tilde`)
- `consistent-emphasis-style` (style: `consistent` | `asterisk` | `underscore`)
- `consistent-list-marker` (style: `consistent` | `dash` | `asterisk` | `plus`)

## Implementation Notes

`linter.New()` currently does not return an error. The change requires one of:

1. Change `New()` to return `(*Linter, error)` and validate options there
2. Add a separate `linter.Validate(cfg)` step called before `New()` in the app layer

Option 1 is cleaner. The app layer already handles errors from config loading, so propagating a `New()` error fits naturally.
